pisek/jobs/parts/verdicts_eval.py

```diff
@@ -0,0 +1,30 @@
+from typing import Optional
+
+from pisek.env.task_config import TaskConfig, FailMode
+from pisek.jobs.parts.solution_result import Verdict, SUBTASK_SPEC, verdict_always
+
+
+def evaluate_verdicts(
+    config: TaskConfig, verdicts: list[Verdict], expected: str
+) -> tuple[bool, bool, Optional[int]]:
+    mode_quantifier = all if config.fail_mode == FailMode.all else any
+
+    result = True
+    definitive = True
+    breaker = None
+    quantifiers = [all, mode_quantifier]
+    for i, quant in enumerate(quantifiers):
+        oks = list(map(SUBTASK_SPEC[expected][i], verdicts))
+        ok = quant(oks)
+
+        result &= ok
+        definitive &= (
+            (quant == any and ok)
+            or (quant == all and not ok)
+            or (SUBTASK_SPEC[expected][i] == verdict_always)
+        )
+        if quant == all and ok == False:
+            breaker = oks.index(False)
+            break
+
+    return result, definitive, breaker
```

Comment:
> Given how different `any` and `all` behave, wouldn't it be easier to just add an `if` here?
> You could also move `ok = quant(oks)` into the cases and avoid passing functions around entirely.

> I don't see how it simplifies things.

It could be written like this:

```suggestion
def evaluate_verdicts(
    config: TaskConfig, verdicts: list[Verdict], expected: str
) -> tuple[bool, bool, Optional[int]]:
    result = True
    definitive = True
    breaker = None

    modes = [FailMode.all, config.fail_mode]

    for i, mode in enumerate(modes):
        oks = list(map(SUBTASK_SPEC[expected][i], verdicts))

        if mode == FailMode.all:
            ok = all(oks)

            result &= ok
            definitive &= not ok or SUBTASK_SPEC[expected][i] == verdict_always

            if not ok:
                breaker = oks.index(False)
                break
        elif mode == FailMode.any:
            ok = any(oks)

            result &= ok
            definitive &= ok

    return result, definitive, breaker
```

This would make it clearer how each mode is checked. One downside is that `result &= ok` is duplicated.